ledger-service/http-json/src/main/scala/com/digitalasset/http/endpoints/MeteringReportEndpoint.scala

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+package endpoints
+
+import akka.http.scaladsl.model.HttpRequest
+import cats.implicits.toTraverseOps
+import com.daml.http.Endpoints.ET
+import com.daml.http.EndpointsCompanion.{Error, ServerError}
+import com.daml.http.endpoints.MeteringReportEndpoint.{
+  MeteringReport,
+  MeteringReportDateRequest,
+  toMeteringReport,
+  toPbRequest,
+}
+import com.daml.http.util.Logging.{InstanceUUID, RequestID}
+import com.daml.jwt.domain.Jwt
+import com.daml.ledger.api.v1.admin.metering_report_service
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Ref.{ApplicationId, ParticipantId}
+import com.daml.lf.data.Time.Timestamp
+import com.daml.logging.LoggingContextOf
+import com.google.protobuf
+import scalaz.std.scalaFuture._
+import scalaz.{-\/, \/, \/-}
+import spray.json.{JsString, JsValue, RootJsonFormat}
+
+import java.time.{Instant, LocalDate, LocalTime, ZoneOffset}
+import scala.concurrent.{ExecutionContext, Future}
+
+private[http] object MeteringReportEndpoint {
+
+  // These classes must use field names that match the Json fields defined in
+  // https://docs.google.com/document/d/1HQyVLCANqw_l_gbxBzgOqsiwmIDNlwCRQffVYJnYTeU/edit#heading=h.1mjra845v4le
```

Comment:
I don't think you should link to a private resource in the open source repo. This format is part of the metering report as part of the open source participant server. This format should be documented as part of it and the reference to this should be accessible to anybody reading this code.

Reply:
In this case the format is also described in public documentation https://docs.daml.com/2.0.0/ops/metering.html which makes a better link than the google doc.